src/Control/Distributed/Spark/RDD.hs

```diff
@@ -79,6 +91,25 @@ map clos rdd = do
     f <- reflect clos
     call rdd "map" [coerce f]
 
+mapPartitions
+  :: (Reflect (Closure (Int32 -> Stream (Of a) IO () -> Stream (Of b) IO ())) ty, Typeable a, Typeable b)
+  => Choice "preservePartitions"
+  -> Closure (Stream (Of a) IO () -> Stream (Of b) IO ())
+  -> RDD a
+  -> IO (RDD b)
+mapPartitions preservePartitions clos rdd =
+  mapPartitionsWithIndex preservePartitions (closure (static const) `cap` clos) rdd
```

Comment:
This would be the version with simpler constraints:
```Haskell
mapPartitions
  :: (Reflect (Closure (Stream (Of a) IO () -> Stream (Of b) IO ())) ty)
  => Choice "preservePartitions"
  -> Closure (Stream (Of a) IO () -> Stream (Of b) IO ())
  -> RDD a
  -> IO (RDD b)
mapPartitions preservePartitions clos rdd = do
  f <- reflect clos
  call rdd "mapPartitions" [coerce f, coerce (Choice.toBool preservePartitions)]
```
but it requires reflecting `Iterable`.